app/assets/javascripts/discourse/lib/utilities.js.es6

```diff
@@ -76,8 +76,8 @@ export function emailValid(email) {
 }
 
 export function urlValid(url) {
-  // see: https://mathiasbynens.be/demo/url-regex
-  var re = /^https?:\/\/(-\.)?([^\s/?\.#-]+\.?)+(\/[^\s]*)?$/i;
+  // adapt from npm: url-regex
+  var re = /^(?:(?:http|https):\/\/)(?:\S+(?::\S*)?@)?(?:(?:(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[0-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]+-?)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]+-?)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))|localhost)(?::\d{2,5})?(?:(\/|\?|#)[^\s]*)?$/i;
```

Comment:
@S-YOU No, it only matches the IPv4 url. I mean any validation should go with state machine to parse that accordingly. Regexp are always a hack but it's simple.

I'll split them.